cloudevents/working-drafts/xml-format.md

```diff
@@ -0,0 +1,243 @@
+
+* The [Envelope](#4-envelope) section defines an XML container for CloudEvents
+attributes and an associated media type.
+
+* The [Batch](#5-xml-batch-format) section describes how multiple CloudEvents
+can be packaged into a single XML document.
+
+### 1.1. Conformance
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC2119][rfc2119].
+
+### 1.2 Approach
+
+This specification is deliberately verbose in favor of readability and
+deterministic processing.
+
+Preservation of attribute value type information is supported allowing
+custom extensions to be communicated without value type loss.
+
+A schema-less approach has been taken favoring convention over rigid document
+structure.
+
+## 2. Attributes
+
+The CloudEvents type system is mapped to the XML schema types as follows :
+
+| CloudEvents   |  XML Schema Type | Notes |
+| :-------------| :--------------- | :---- |
+| Boolean       | [xs:boolean][xml-primitives] | |
+| Integer       | [xs:int][xml-primitives] | |
+| String        | [xs:string][xml-primitives] | |
+| Binary        | [xs:base64Binary][xml-primitives] | |
+| URI           | [xs:anyURI][xml-primitives] following [RFC 3986][rfc3986]| |
+| URI-reference | [xs:anyURI][xml-primitives] following [RFC 3986][rfc3986] | |
+| Timestamp     | [xs:dateTime][xml-primitives] following [RFC 3339][rfc3339] | |
+
+Each context attribute is represented as an XML element whose name MUST exactly match
+that of a REQUIRED or OPTIONAL CloudEvent attribute, extension attribute names MUST
+adhere to, and align with, the conventions in defined in the [cloud event specification][ce-spec].
+
+See the [envelope](#4-envelope) for special handing of the `specversion` context attribute.
+
+Extension attributes SHOULD be expressed with the appropriate `xsi:type` to allow them
+to be exchanged without loss of type information. Extension attributes with no `xsi:type`
+discriminator SHOULD be interpreted with an implied type of `xs:string`
+
+``` xml
+    ...
+    <id>AAABBBCCCNNN0000</id>
+    ..
+    <time>2021-08-14T-08:00</time>
+    ..
+    <myextension xsi:type="xs:string">my extension value</myextension>
+    <myboolean xsi:type="xs:boolean">false</myboolean>
+```
+
+## 3. Data
+
+The `Data` portion of a CloudEvent follows a similar model to that employed by
+the [JSON Format specification][json-format]. The element names MUST be capitalized to avoid
+collision with curent or future CloudEvent attribute names.
+
+### 3.1 Binary Data
+
+MUST be carried in an element with an defined type of `xs:base64Binary`
+
+``` xml
+<Data xsi:type="xs:base64Binary">.........</Data>
+```
+
+### 3.2 Text Data
+
+MUST be carried in an element with an defined type of `xs:string`
+
+``` xml
+<Data xsi:type="xs:string">This is text</Data>
+```
+
+### 3.3 XML Data
+
+XML data MUST be carried in an explicit element `<DataXml>`:
```

Comment:
I think a bit more detail would be useful here. Is the intention that the content of `DataXml` is a single element? Could it just be multiple elements? Can it be empty? If it's always a single element, is the whitespace around the element considered significant? It might be worth having examples of everything that's possible.

(Oh, and clarify that attributes are not permitted on the `DataXml` element itself, assuming that's what we want?)